nobl9/resource_budgetadjustment.go

```diff
@@ -0,0 +1,289 @@
+				"The expected value for this field is a string formatted as a time duration. " +
+				"The duration must be defined with a precision of 1 minute. " +
+				"Example: `1h10m`",
+		},
+		"rrule": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: validateRrule,
+			Description: "The recurrence rule for the budget adjustment event. " +
+				"The expected value is a string in RRULE format. " +
+				"Example: `FREQ=MONTHLY;BYMONTHDAY=1`",
+		},
+		"filters": {
+			Type:        schema.TypeSet,
+			Optional:    true,
+			Description: "Filters are used to select SLOs for the budget adjustment event.",
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"slos": {
+						Type:     schema.TypeSet,
+						Required: true,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"slo": {
+									Type:        schema.TypeList,
+									MinItems:    1,
+									Required:    true,
+									Description: "SLO where budget adjustment event will be applied.",
+									Elem: &schema.Resource{
+										Schema: map[string]*schema.Schema{
+											"name": {
+												Type:        schema.TypeString,
+												Required:    true,
+												Description: "Unique name of the resource, must conform to the naming convention from [DNS RFC1123](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names).",
+											},
+											"project": {
+												Type:        schema.TypeString,
+												Required:    true,
+												Description: "Name of the Nobl9 project the resource sits in, must conform to the naming convention from [DNS RFC1123](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names).",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func validateDuration(v interface{}, path cty.Path) diag.Diagnostics {
+	var diags diag.Diagnostics
+	_, err := time.ParseDuration(v.(string))
+	if err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity:      diag.Error,
+			Summary:       "Invalid duration format",
+			Detail:        fmt.Sprintf("Invalid duration format: %s", v),
+			AttributePath: path,
+		})
+	}
+	return diags
+}
+
+func validateRrule(v interface{}, path cty.Path) diag.Diagnostics {
+	var diags diag.Diagnostics
+	_, err := rrule.StrToRRule(v.(string))
+	if err != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity:      diag.Error,
+			Summary:       "Invalid rrule format",
+			Detail:        fmt.Sprintf("Invalid rrule format: %s", v),
+			AttributePath: path,
+		})
+	}
+	return diags
+}
+
+func resourceBudgetAdjustmentApply(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(ProviderConfig)
+	client, ds := getClient(config)
+	if ds != nil {
+		return ds
+	}
+
+	budgetAdjustment := marshalBudgetAdjustment(d)
```

Comment:
shadows budgetAdjustment() function, maybe just (or `bg`):

```suggestion
	adjustment := marshalBudgetAdjustment(d)
```